src/classes/particle.ts

```diff
@@ -366,48 +251,181 @@ export class Particle {
         }
     }
 
-    public bubble() {
+    public bubble(): void {
         this.bubbler.bubble();
     }
 
-    public repulse() {
+    public repulse(): void {
         this.repulser.repulse();
     }
 
     /* ---------- tsParticles functions - particles interaction ------------ */
-    public link(p2: Particle) {
+    public link(p2: Particle): void {
         this.updater.link(p2);
     }
 
-    public attract(p2: Particle) {
+    public attract(p2: Particle): void {
         this.updater.attract(p2);
     }
 
-    public bounce(p2: Particle) {
+    public bounce(p2: Particle): void {
         this.updater.bounce(p2);
     }
 
-    public move(delta: number) {
+    public move(delta: number): void {
         this.updater.move(delta);
     }
 
-    public moveParallax() {
+    public moveParallax(): void {
         this.updater.moveParallax();
     }
 
-    public updateOpacity() {
+    public updateOpacity(): void {
         this.updater.updateOpacity();
     }
 
-    public updateSize() {
+    public updateSize(): void {
         this.updater.updateSize();
     }
 
     public fixOutOfCanvasPosition() {
         this.updater.fixOutOfCanvasPosition();
     }
 
-    public updateOutMode() {
+    public updateOutMode(): void {
         this.updater.updateOutMode();
     }
+
+    private calcPosition(container: Container, position?: ICoordinates): ICoordinates {
+        let pos = {
+            x: position && position.x ? position.x : Math.random() * container.canvas.w,
+            y: position && position.y ? position.y : Math.random() * container.canvas.h,
+        };
+
+        /* check position  - into the canvas */
+        if (pos.x > container.canvas.w - this.radius * 2) {
+            pos.x -= this.radius;
+        } else if (pos.x < this.radius * 2) {
+            pos.x += this.radius;
+        }
+
+        if (pos.y > container.canvas.h - this.radius * 2) {
+            pos.y -= this.radius;
+        } else if (pos.y < this.radius * 2) {
+            pos.y += this.radius;
+        }
+
+        return pos;
+    }
+
+    private calcVelocity(options: IOptions): IVelocity {
+        const velbase = this.getVelBase(options);
+        let res = {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Identifier 'res' is never reassigned; use 'const' instead of 'let'.](https://app.codacy.com/manual/ar3s/tsparticles/pullRequest?prid=5011230)